### DISCUSSION TITLE
Make API response data simple to manage without code and usable within app

### DISCUSSION THREAD

**Discussion by rpmcoding** [Budibase/budibase#7927] [Ideas]
Budibase is really great.  However, when it comes to handling data returned by an API call, you really do have to start coding and use the data transformer.  And it's not easy for us non-coders.

To make the product truly no code, I'd suggest the json data returned is automatically parsed into a usable array across the app which can then be accessed via dropdown selection fields.  No code involved at all.

**Comment by melohagan** [comment]
Hey @rpmcoding 

I think the data transformer will always be needed in some form. We simply don't know what format data will be returned from any given API. 
If you don't want to use the data transformer then the datasources should help with that. In particular, the newly released [custom datasource plugins](https://github.com/Budibase/plugins) could be built to help non-coding users get the data in a format that doesn't need to be transformed.
